src/Compression/CompressionCodecEncrypted.h

```diff
@@ -79,16 +85,19 @@ namespace DB
 
     private:
         static std::string lastErrorString();
-        static std::string deriveKey(const std::string_view & master_key);
         static void encrypt(const std::string_view & plaintext, char * ciphertext_and_tag);
-        static void decrypt(const std::string_view & ciphertext_and_tag, char * plaintext);
+        static void decrypt(const std::string_view & ciphertext_and_tag, char * plaintext, UInt64 key_id=current_key_id);
```

Comment:
Why `encrypt`, `decrypt` and `getKeys()` are static? There can be lots of threads doing encryption/decryption at the same time with different keys.

Reply:
`encrypt` and `decrypt` are called from `doCompressData` and `doDecompressData`, which are not static and have no problem with calling them from different threads.